addons/website/controllers/manual_models.py

```diff
@@ -0,0 +1,122 @@
+import ast
+
+import werkzeug
+
+from odoo.addons.http_routing.models.ir_http import slug, unslug
+from odoo.http import Controller, request, route
+from odoo.osv.expression import AND
+
+
+def get_relevant_field_among(names, fields):
+    return next(iter([f for f in names if f in fields]))
+
+class ManualModelsController(Controller):
+
+    @route([
+        "/model/<string:page_name_slugified>",
+        "/model/<string:page_name_slugified>/page/<int:page_number>",
+        "/model/<string:page_name_slugified>/<string:record_slug>",
+    ], website=True, auth="public")
+    def generic_manual_model(self, page_name_slugified=None, page_number=1, record_slug=None, **searches):
+        if not page_name_slugified:
+            raise werkzeug.exceptions.NotFound()
+
+        env = request.env
+        website = request.website
+
+        page_type = "listing"
+        if record_slug is not None:
+            page_type = "single"
+
+        website_page_domain = AND([
+            [("page_type", "=", page_type)],
+            [("name_slugified", "=", page_name_slugified)],
+            website.website_domain(),
+        ])
+
+        page = env["website.controller.page"].sudo().search(website_page_domain, limit=1)
+        if not page:
+            raise werkzeug.exceptions.NotFound()
+
+        if not page.model_id.state == "manual":
+            raise werkzeug.exceptions.NotFound()
+
+        view = page.view_id
+        if not view:
+            raise werkzeug.exceptions.NotFound()
+
+        target_model_name = page.model_id.model
+        model_sudo = env[target_model_name].sudo()
+        rec_domain = ast.literal_eval(page.record_domain or "[]")
+
+        fields = model_sudo._fields
+        domains = [rec_domain]
+        if "website_id" in fields:
+            domains.append(website.website_domain())
+        if "website_published" in fields and not request.env.user.has_group('website.group_website_designer'):
+            domains.append([("website_published", "=", True)])
+
+        if record_slug:
+            _, res_id = unslug(record_slug)
+            record = model_sudo.browse(res_id).filtered_domain(AND(domains))
+            # We check for slug matching because we are not entirely sure
+            # that we end up seeing record for the right model
+            # i.e. in case of a redirect when a "single" page doesn't match the listing
+            if not record.exists() or record_slug != slug(record):
+                raise werkzeug.exceptions.NotFound()
+
+            render_context = {"record": record}
+            return request.render(view.key, render_context)
+
+        layout_mode = request.session.get(f'website_studio_{view.id}_layout_mode')
+        if not layout_mode:
+            # use the default layout set for this page
+            layout_mode = request.env["website.controller.page"]._get_default_layout(view.id)
```

Comment:
this method and the field behind it should be defined here, in community (for now it is defined in website_studio)
Being able to easily customize this feature is a Studio feature, but in this task *using* the feature layout is a community one